pkg/controller/changetierrequest/changetierrequest_controller_test.go

```diff
@@ -0,0 +1,345 @@
+	"github.com/spf13/cast"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	apiv1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestChangeTierSuccess(t *testing.T) {
+	// given
+	restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_DURATION_BEFORE_CHANGE_REQUEST_DELETION", "10s")
+	defer restore()
+	teamTier := NewNSTemplateTier("team", "123team", "stage", "dev")
+	basicTier := NewNSTemplateTier("basic", "123abc", "code", "stage", "dev")
+
+	t.Run("the controller should change tier in MUR", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("john")
+		changeTierRequest := newChangeTierRequest("john", "team", "", noStatus)
+		controller, request, cl := newController(t, changeTierRequest, mur, teamTier)
+
+		// when
+		_, err := controller.Reconcile(request)
+
+		// then
+		require.NoError(t, err)
+		murtest.AssertThatMasterUserRecord(t, "john", cl).
+			AllUserAccountsHaveTier(*teamTier)
+		AssertThatChangeTierRequestHasCondition(t, cl, changeTierRequest.Name, toBeComplete())
+	})
+
+	t.Run("the controller should change tier in all UserAccounts in MUR", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("johny", murtest.AdditionalAccounts("another-cluster"))
+		changeTierRequest := newChangeTierRequest("johny", "team", "", noStatus)
+		controller, request, cl := newController(t, changeTierRequest, mur, teamTier)
+
+		// when
+		_, err := controller.Reconcile(request)
+
+		// then
+		require.NoError(t, err)
+		murtest.AssertThatMasterUserRecord(t, "johny", cl).
+			AllUserAccountsHaveTier(*teamTier)
+		AssertThatChangeTierRequestHasCondition(t, cl, changeTierRequest.Name, toBeComplete())
+	})
+
+	t.Run("the controller should change tier only in specified UserAccount in MUR", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("johny", murtest.AdditionalAccounts("another-cluster"))
+		changeTierRequest := newChangeTierRequest("johny", "team", "another-cluster", noStatus)
+		controller, request, cl := newController(t, changeTierRequest, mur, teamTier)
+
+		// when
+		_, err := controller.Reconcile(request)
+
+		// then
+		require.NoError(t, err)
+		murtest.AssertThatMasterUserRecord(t, "johny", cl).
+			UserAccountHasTier("another-cluster", *teamTier).
+			UserAccountHasTier(test.MemberClusterName, *basicTier)
+		AssertThatChangeTierRequestHasCondition(t, cl, changeTierRequest.Name, toBeComplete())
+	})
+
+	t.Run("will not do anything and return requeue with shorter duration that 10s", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("johny")
+		changeTierRequest := newChangeTierRequest("johny", "team", "", noStatus)
```

Comment:
what about refactoring the `newChangeTierRequest` func to a variadic func, where the last param would be the conditions (and if none is given, set to `noStatus`?)

Reply:
actually, we don't need to pass the status at all - we don't use it anywhere. It's probably some zombie from previous versions :ghost: fixed in 8280048

Reply:
oh, ok. No need for such a change indeed, then.